js/Common.js

```diff
@@ -1246,7 +1246,7 @@ if (!CRM.vars) CRM.vars = {};
         unique: true
       };
       options = $.extend(extra, options);
-      options.expires = options.expires === false ? 0 : parseInt(options.expires, 10);
+      options.expires = (options.expires === false || (!CRM.config.dismissInfoAlerts && type == "info")) ? 0 : parseInt(options.expires, 10);
```

Comment:
There seems to be a logic error here. Automatic dismissal should be prevented for all errors, not just Info errors. I think you need to remove 
&& type == "info"